var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -52,10 +56,27 @@ class Mesa(AutotoolsPackage):
     depends_on('libxfixes')
 
     depends_on('glproto@1.4.14:', type='build')
-    depends_on('dri2proto@2.6:', type='build')
-    depends_on('dri3proto@1.0:', type='build')
+    depends_on('libdrm', when='+dri')
+    depends_on('dri2proto@2.6:', type='build', when='+dri')
+    depends_on('dri3proto@1.0:', type='build', when='+dri')
+    depends_on('expat', when='+dri')
+    depends_on('llvm@3:3.99', when='+dri')
```

Comment:
i think it needs just-in-time compiler for shaders or something along these lines. So no, i think it can't